nixos/modules/services/misc/taskserver/helper-tool.py

```diff
@@ -90,7 +94,7 @@ def certtool_cmd(*args, **kwargs):
     """
     return subprocess.check_output(
         [CERTTOOL_COMMAND] + list(args),
-        preexec_fn=lambda: os.umask(0o077),
```

Comment:
Does that imply that setting the `umask` doesn't work here?

Reply:
> Does that imply that setting the `umask` doesn't work here (eg. `certtool` ignores it)?

Pretty much yes. `certtool` ensures by itself that only the user running `certtool` will have read and write access to the private keys that are generated. Hence the `chmod` for adding group read access to the private key is needed.

Reply:
Ah, okay... seems reasonable then to do the `chmod`. I'm however never used the `taskserver` module myself (stopped using taskwarrior directly afterwards) even though I implemented it, so maybe I'm not the best one to review this properly. From looking at it however the change looks fine to me.

Reply:
TBH, I don't use taskserver myself as well anymore. I just tested the NixOS module for a brief period and it did work for me during that period. Since this PR doesn't change anything fundamental, I think it'd be safe enough to merge it. But it's your call. We can also ask for a review in one of the discourse threads [1](https://discourse.nixos.org/t/prs-in-distress/3604/60) or [2](https://discourse.nixos.org/t/prs-ready-for-review/3032/485).

Reply:
Maybe @arcnmx, @pacien or @Pamplemousse can help there, since they have made changes to the module in recent years. If noone objects I'd merge in a few days since I don't think this change will cause a big disruption.

Reply:
I sadly don't have time to test this, but LGTM :+1: